Appwrite migration fixes

xkaper001 opened the pull request:
# Summary

Fixes the following features:
- auth improvements
- bookmarking the monument
- model enhancements
- script fixes

## Description

This pull request includes several changes to the authentication system, data models, and repository setup to integrate Appwrite and improve data handling. The most important changes include switching from Firebase to Appwrite for authentication, updating data models to include new fields, and enhancing the Appwrite setup script.

### Authentication System Updates:
* Replaced Firebase authentication with Appwrite be default in `login_register_bloc.dart` and updated the exception handling accordingly. [[1]](diffhunk://#diff-5854c0877fd9d3be475e83e2a50e93f8fc072eeb71ee862c11d486b715fc437fL4-R4) [[2]](diffhunk://#diff-5854c0877fd9d3be475e83e2a50e93f8fc072eeb71ee862c11d486b715fc437fL163-R163)
* Modified the `AppwriteAuthenticationRepository` to use the correct Appwrite document ID field (`$id`) instead of `uid`. [[1]](diffhunk://#diff-b97e0e947bc87ab37633cf91e246decaad019518f6e361f6e0f3e46c5834cd0bL92-R92) [[2]](diffhunk://#diff-b97e0e947bc87ab37633cf91e246decaad019518f6e361f6e0f3e46c5834cd0bL109) [[3]](diffhunk://#diff-b97e0e947bc87ab37633cf91e246decaad019518f6e361f6e0f3e46c5834cd0bL225-R224) [[4]](diffhunk://#diff-b97e0e947bc87ab37633cf91e246decaad019518f6e361f6e0f3e46c5834cd0bL239)

### Data Model Enhancements:
* Updated `MonumentModel` and `UserModel` to use the Appwrite document ID field (`$id`) and added new fields for saved and checked-in monuments. [[1]](diffhunk://#diff-d3a7b616664d466dda05cc900a6b9652afb65183f664e40f747e94725b72d845R10) [[2]](diffhunk://#diff-d74913f8f37af34e661482840b54a40b6f55a7b584f7958b9f9aa30bc36118edL16-R16) [[3]](diffhunk://#diff-d74913f8f37af34e661482840b54a40b6f55a7b584f7958b9f9aa30bc36118edL37-R37) [[4]](diffhunk://#diff-42e953b1e31f1a2dd01b1dcaf41f9f2a173e68ec1af4efc07ddbb8b3c0c07f0dR2-R9) [[5]](diffhunk://#diff-42e953b1e31f1a2dd01b1dcaf41f9f2a173e68ec1af4efc07ddbb8b3c0c07f0dR19-R27) [[6]](diffhunk://#diff-42e953b1e31f1a2dd01b1dcaf41f9f2a173e68ec1af4efc07ddbb8b3c0c07f0dL76-R82) [[7]](diffhunk://#diff-42e953b1e31f1a2dd01b1dcaf41f9f2a173e68ec1af4efc07ddbb8b3c0c07f0dR91-R106) [[8]](diffhunk://#diff-42e953b1e31f1a2dd01b1dcaf41f9f2a173e68ec1af4efc07ddbb8b3c0c07f0dR115-R116) [[9]](diffhunk://#diff-c15d360f1150d66e50f43c3c37eafed5b519ead0ded251226fc262634eb9ead9R21-R38) [[10]](diffhunk://#diff-c15d360f1150d66e50f43c3c37eafed5b519ead0ded251226fc262634eb9ead9R47-R49)
* Updated `UserEntity` to reflect changes in the data model and use Appwrite's `Document` instead of Firebase's `DocumentSnapshot`. [[1]](diffhunk://#diff-be519d8a35d9c8ac384a0b120a756d3c05bea27e006b50a1d23114118a772a11L3-R3) [[2]](diffhunk://#diff-be519d8a35d9c8ac384a0b120a756d3c05bea27e006b50a1d23114118a772a11L57-R58) [[3]](diffhunk://#diff-be519d8a35d9c8ac384a0b120a756d3c05bea27e006b50a1d23114118a772a11L70-R70) [[4]](diffhunk://#diff-be519d8a35d9c8ac384a0b120a756d3c05bea27e006b50a1d23114118a772a11L86-R86)

### Repository Setup Improvements:
* Enhanced the `setup_appwrite.js` script to include the creation of a storage bucket for images and added necessary permissions. [[1]](diffhunk://#diff-245bc112a42cb2628781836193b17ae2609249f4b7f8f7a0c4ca24cc581fb086L1-R1) [[2]](diffhunk://#diff-245bc112a42cb2628781836193b17ae2609249f4b7f8f7a0c4ca24cc581fb086L30-R30) [[3]](diffhunk://#diff-245bc112a42cb2628781836193b17ae2609249f4b7f8f7a0c4ca24cc581fb086R43) [[4]](diffhunk://#diff-245bc112a42cb2628781836193b17ae2609249f4b7f8f7a0c4ca24cc581fb086R56-R58) [[5]](diffhunk://#diff-245bc112a42cb2628781836193b17ae2609249f4b7f8f7a0c4ca24cc581fb086R70-R97) [[6]](diffhunk://#diff-245bc112a42cb2628781836193b17ae2609249f4b7f8f7a0c4ca24cc581fb086R121-R125) [[7]](diffhunk://#diff-245bc112a42cb2628781836193b17ae2609249f4b7f8f7a0c4ca24cc581fb086L157-R201)

### Miscellaneous Changes:
* Updated the `load-env.gradle` script to handle environment variable parsing more robustly.
* Improved error messaging in the `login_view_mobile.dart` file to display dynamic error messages from the state.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor (does not change functionality, e.g., code style improvements, linting)

xkaper001 committed:
fix bugs in authentication and data models

xkaper001 committed:
Enhance Appwrite setup script to include storage bucket creation and update permissions

xkaper001 committed:
fix bookmark feature bugs

xkaper001 committed:
Update .env.template to set APPWRITE_BUCKET_ID for the monument bucket

xkaper001 committed:
comment out registration for FirebaseMonumentRepository in service locator and switch to appwrite by default

xkaper001 commented:
@mdmohsin7 @AyaNady17 pls review this.

AyaNady17 commented:
Hey @xkaper001 ,  

Let's keep this PR focused on fixing the Appwrite script issues, the permissions part.  
For authentication, it's already working, no need for changes RN, and regarding bookmarking monuments  and model enhancements, I’m not sure what issues you're addressing or the purpose of the changes.  

Please keep each PR specific to one issue. If you're fixing something else, kindly state the problem and solution in a separate PR.  
Thanks!

xkaper001 commented:
Hey @AyaNady17,

The changes in the authentication part are just an optimization/enhancement to use `$id` in place of `uid`. Also regarding the bookmarking feature, it was not working; it's broken. So that's what I have fixed in this PR. 

And regarding model enhancements, that has been done in accordance with the Appwrite script to help further the implementation of other features such as bookmarking and check-ins. 

That's all I have addressed in this PR – what's needed – and it doesn't include any changes that aren't related.

AyaNady17 commented:
Now we're dealing with three different changes in one branch. To keep things cleaner and more manageable for review, it's best to separate them. This helps ensure that we don’t merge any uncertain code.  
Let’s do the following:  
- I don’t see an optimization in changing `uid` to `$id`—`uid` is more meaningful in this context so let’s keep it.  
- Keep this PR focused **only on fixing the permissions issue**.  
- For the **broken bookmarking feature**, please open a separate PR specifically for that and attach **before-and-after videos** if possible.  

This way, everything stays **organized and easy to review**. Thanks!

xkaper001 commented:
okay works. will do that.

regarding `uid` and `$id` that's just the tip of the iceberg you're noticing. Actually a complete attribute field has been removed from the Appwrite database, as we could work with the document id, i.e. $id, instead. and use of `uid` is redundant.

AyaNady17 commented:
> okay works. will do that.
> 
> regarding `uid` and `$id` that's just the tip of the iceberg you're noticing. Actually a complete attribute field has been removed from the Appwrite database, as we could work with the document id, i.e. $id, instead. and use of `uid` is redundant.

If keeping it is not causing an actual issue right now, let’s not waste time debating it.
I'll close this PR as you created another one.
Thanks @xkaper001
